mdata/store_cassandra.go

```diff
@@ -73,13 +76,68 @@ It's safe for concurrent use by multiple goroutines and a typical usage scenario
 object to interact with the whole Cassandra cluster.
 */
 
+type ttlTables map[uint32]ttlTable
+type ttlTable struct {
+	table  string
+	window uint32
+}
+
 type cassandraStore struct {
 	session     *gocql.Session
 	writeQueues []chan *ChunkWriteRequest
 	readQueue   chan *ChunkReadRequest
+	ttlTables   ttlTables
+}
+
+func ttlUnits(ttl uint32) float64 {
+	// convert ttl to hours
+	return float64(ttl) / (60 * 60)
+}
+
+func getTTLTables(ttls []uint32, windowFactor int, nameFormat string) ttlTables {
+	tables := make(ttlTables)
+	for _, ttl := range ttls {
+		/*
+		 * the purpose of this loop is to bucket metrics of similar TTLs.
+		 * we first calculate the largest power of 2 that's smaller than the TTL and then divide the result by
+		 * the window factor. for example with a window factor of 20 we want to group the metrics like this:
+		 *
+		 * generated with: https://gist.github.com/replay/69ad7cfd523edfa552cd12851fa74c58
+		 *
+		 * +-----------------------------+---------------+-------------+----------+
+		 * |                hours / days |    table_name | window_size | sstables |
+		 * +-----------------------------+---------------+-------------+----------+
+		 * |                      1 / <1 |     metrics_1 |           1 |    1 - 1 |
```

Comment:
>  Do you mean the values between the limits of the different rows?

yes. I think your suggested table is basically the same as my suggestion, but I still see two slight advantages to my proposal:

* seeing `floor` in the header and figuring out how that relates to the values shown (take your ttl, floor it, then look at the table) is not immediately obvious. though you could clarify that with a comment instructing people how to use the table, but I think my version is clearer more quickly.
* my values are simply all the powers of 2, and they relate directly to the table names in the 2nd column. Values like 31 and 1023 look slightly weird.

> Actually I think an approximation is good enough in this case, otherwise it just gets unnecessarily complicated

it seems trivial to me to do the right thing here, no need for approximations. your version isn't an approximation either (which is good) so I don't understand what you're getting at here.